clang/include/clang/3C/ConstraintVariables.h

```diff
@@ -351,11 +351,15 @@ class PointerVariableConstraint : public ConstraintVariable {
   PointerVariableConstraint(CAtoms V, std::vector<ConstAtom *> SV,
                             std::string T, std::string Name,
                             FunctionVariableConstraint *F, std::string Is,
-                            int Generic = -1)
-      : ConstraintVariable(PointerVariable, "" /*not used*/, Name), BaseType(T),
+                            int Generic=-1, bool IsTypeddef=false, TypedefNameDecl *TDT=nullptr,
+                            std::string TypedefString="",
+                            struct InternalTypedefInfo TypedefLevelInfo={}) :
+    ConstraintVariable(PointerVariable, "" /*not used*/, Name), BaseType(T),
         Vars(V), SrcVars(SV), FV(F), SrcHasItype(!Is.empty()), ItypeStr(Is),
         PartOfFuncPrototype(false), Parent(nullptr), BoundsAnnotationStr(""),
-        GenericIndex(Generic), IsZeroWidthArray(false), IsVoidPtr(false) {}
+        GenericIndex(Generic), IsZeroWidthArray(false), IsTypedef(IsTypeddef),
+        TDT(TDT), TypedefString(TypedefString),
+        TypedefLevelInfo(TypedefLevelInfo), IsVoidPtr(false) {}
```

Comment:
Maybe it's clear to everyone else, but can you explain why all these extra field initializations are being added in this PR?

Reply:
This constructor previously left these fields uninitialized. The initialized values were later read in `mkString`.

Reply:
So this is an unrelated bug fix?  Is there an easy way to write a test for it, or does a test that you're already adding cover it?

Reply:
I found these converting Icecast under valgrind while investing the issue fixed by this PR. They didn't turn out to be related, and I don't have any specific test case for them.

Reply:
Fine with me then.